base/lthooks.dtx

```diff
@@ -5137,6 +6135,53 @@
       Please~ use~ a~ different~ name~ for~ your~ hook.}
 %    \end{macrocode}
 %
+%    \begin{macrocode}
+%<latexrelease>\IncludeInRelease{2023/06/01}{too-many-args}
+%<latexrelease>                 {Hooks~with~args}
+%    \end{macrocode}
+%
+%    \begin{macrocode}
+\msg_new:nnnn { hooks } { too-many-args }
+  { Too~many~arguments~for~hook~'#1'. }
+  {
+    You~tried~to~declare~a~hook~with~#2~arguments,~but~a~
+    hook~can~only~have~up~to~nine.~LaTeX~will~define~this~
+    hook~with~nine~arguments.
+  }
+%    \end{macrocode}
+%
+%    \begin{macrocode}
+\msg_new:nnnn { hooks } { without-args }
+  { Hook~'#1'~has~no~arguments. }
+  {
+    You~tried~to~use~\iow_char:N\\#2WithArguments~
+    on~a~hook~that~takes~no~arguments.\\
+    Check~the~usage~of~the~hook~or~use~\iow_char:N\\#2~instead.\\
+    \\
+    LaTeX~will~use~\iow_char:N\\#2.
+  }
+%    \end{macrocode}
+%
+%    \begin{macrocode}
+\msg_new:nnnn { hooks } { one-time-args }
+  { You~can't~have~arguments~in~used~one-time~hook~'#1'. }
+  {
+    You~tried~to~use~\iow_char:N\\#2WithArguments~
+    on~a~one-time~hook~that~has~already~been~used.~
+    You~have~to~add~the~code~before~the~hook~is~used,~
+    or~add~the~code~without~arguments~using~\iow_char:N\\#2~instead.\\
```

Comment:
this reads a bit strangely ...wouldn't understand it I guess

Reply:
What about this?
```
  { Arguments~not~allowed~in~used~one-time~hook~'#1'. }
  {
    You~can't~have~arguments~in~a~one-time~hook~that~
    has~already~been~used. ~You~have~to~add~the~code~
    earlier,~or~use~\iow_char:N\\#2~instead.
  }
```
(`#2` is `\AddToHook` or `\AddToHookNext`)

Reply:
Not sure about the main error text... I was thinking of something like
```
  { One-time hook #1 already used -- can't add code with arguments any longer }
  { You can't use \...WithArguments  on a one-time hook that was already executed. ....
```